src/pilot/assign_role.py

```diff
@@ -296,48 +296,120 @@ def get_raid_controller_ids(drac_client):
     return raid_controller_ids
 
 
-def define_controller_logical_disks(drac_client, raid_controller_ids):
-    if len(raid_controller_ids) > 1:
-        LOG.critical(
-            "Found more than one RAID controller:\n  {}".format(
-                "\n  ".join(raid_controller_ids)))
-        return None
+def define_controller_logical_disks(drac_client, raid_controllers):
+    all_physical_disks = drac_client.list_physical_disks()
 
-    raid_10_logical_disk = define_single_raid_10_logical_disk(
-        drac_client, raid_controller_ids[0])
+    # Get the drives controlled by the RAID controller
+    raid_cntlr_physical_disks = {}
+    for disk in all_physical_disks:
+        if disk.controller in raid_controllers:
+            if disk.controller in raid_cntlr_physical_disks:
+                raid_cntlr_physical_disks[disk.controller].append(disk)
+            else:
+                raid_cntlr_physical_disks[disk.controller] = [disk]
 
-    # None indicates an error occurred.
-    if raid_10_logical_disk is None:
+    boss_controller = [cntrl for cntrl in raid_controllers
+                       if drac_client.is_boss_controller(cntrl)]
+
```

Comment:
Perhaps add new function _fetch_boss_controller is cleaner than this construct, also doesn't this return a array so for it to be correct it would be:
[cntrl for cntrl in raid_controller if drac_client.is_boss_controller(cntrl)][0] ?

Reply:
Agree. I will add new function which fetch the boss controller and return None or boss card.

Reply:
Done